### ISSUE TITLE
error during make : For bug reporting instructions, please see: <file:///usr/share/doc/gcc-13/README.Bugs>. ar: providers/common/der/libdefault-lib-der_rsa_sig.o: No such file or directory make[1]: *** [Makefile:22733: providers/libdefault.a] Error 1 make[1]: Leaving directory '/home/alexandre/Téléchargements/openssl' make: *** [Makefile:3731 : build_sw] Erreur 2

### ISSUE THREAD

**Issue by aragon5956** [openssl/openssl#24766]
hello i've this problem during compiling with `make -j8`

```
For bug reporting instructions, please see:
<file:///usr/share/doc/gcc-13/README.Bugs>.
ar: providers/common/der/libdefault-lib-der_rsa_sig.o: No such file or directory
make[1]: *** [Makefile:22733: providers/libdefault.a] Error 1
make[1]: Leaving directory '/home/alexandre/Téléchargements/openssl'
make: *** [Makefile:3731 : build_sw] Erreur 2
```

before i did : `./Configure`

and my configuration is : 

```
alexandre@alexandre-developpeur:~/Téléchargements/openssl$ lsb_release -a
No LSB modules are available.
Distributor ID:	Ubuntu
Description:	Ubuntu 23.10
Release:	23.10
Codename:	mantic
alexandre@alexandre-developpeur:~/Téléchargements/openssl$ uname -r
6.5.0-42-generic
```

what can i do ?
Cordially

**Comment by levitte** [comment]
What do you get if you do this?

``` console
$ cd ~/Téléchargements/openssl
$ ./configdata.pm -d
```

**Comment by aragon5956** [comment]
i succeed it before you message but i don't know why !

`./configdata.pm -d

Command line (with current working directory = .):

    /usr/bin/perl ./Configure

Perl information:

    /usr/bin/perl
    5.36.0 for x86_64-linux-gnu-thread-multi

Enabled features:

    afalgeng
    apps
    argon2
    aria
    asm
    async
    atexit
    autoalginit
    autoerrinit
    autoload-config
    bf
    blake2
    bulk
    cached-fetch
    camellia
    capieng
    cast
    chacha
    cmac
    cmp
    cms
    comp
    ct
    default-thread-pool
    deprecated
    des
    dgram
    dh
    docs
    dsa
    dso
    dtls
    dynamic-engine
    ec
    ec2m
    ecdh
    ecdsa
    ecx
    engine
    err
    filenames
    gost
    http
    idea
    legacy
    loadereng
    makedepend
    md4
    mdc2
    module
    multiblock
    nextprotoneg
    ocb
    ocsp
    padlockeng
    pic
    pinshared
    poly1305
    posix-io
    psk
    quic
    unstable-qlog
    rc2
    rc4
    rdrand
    rfc3779
    rmd160
    scrypt
    secure-memory
    seed
    shared
    siphash
    siv
    sm2
    sm2-precomp
    sm3
    sm4
    sock
    srp
    srtp
    sse2
    ssl
    ssl-trace
    static-engine
    stdio
    tests
    thread-pool
    threads
    tls
    integrity-only-ciphers
    ts
    ui-console
    whirlpool
    tls1
    tls1-method
    tls1_1
    tls1_1-method
    tls1_2
    tls1_2-method
    tls1_3
    dtls1
    dtls1-method
    dtls1_2
    dtls1_2-method

Disabled features:

    acvp-tests             [cascade]        OPENSSL_NO_ACVP_TESTS
    asan                   [default]        OPENSSL_NO_ASAN
    brotli                 [default]        OPENSSL_NO_BROTLI
    brotli-dynamic         [default]        OPENSSL_NO_BROTLI_DYNAMIC
    buildtest-c++          [default]        
    winstore               [not-windows]    OPENSSL_NO_WINSTORE
    crypto-mdebug          [default]        OPENSSL_NO_CRYPTO_MDEBUG
    demos                  [default]        OPENSSL_NO_DEMOS
    h3demo                 [default]        OPENSSL_NO_H3DEMO
    devcryptoeng           [default]        OPENSSL_NO_DEVCRYPTOENG
    ec_nistp_64_gcc_128    [default]        OPENSSL_NO_EC_NISTP_64_GCC_128
    egd                    [default]        OPENSSL_NO_EGD
    external-tests         [default]        OPENSSL_NO_EXTERNAL_TESTS
    fips                   [default]        
    fips-securitychecks    [cascade]        OPENSSL_NO_FIPS_SECURITYCHECKS
    fuzz-afl               [default]        OPENSSL_NO_FUZZ_AFL
    fuzz-libfuzzer         [default]        OPENSSL_NO_FUZZ_LIBFUZZER
    ktls                   [default]        OPENSSL_NO_KTLS
    md2                    [default]        OPENSSL_NO_MD2 (skip crypto/md2)
    msan                   [default]        OPENSSL_NO_MSAN
    pie                    [default]        OPENSSL_NO_PIE
    rc5                    [default]        OPENSSL_NO_RC5 (skip crypto/rc5)
    sctp                   [default]        OPENSSL_NO_SCTP
    tfo                    [default]        OPENSSL_NO_TFO
    trace                  [default]        OPENSSL_NO_TRACE
    ubsan                  [default]        OPENSSL_NO_UBSAN
    unit-test              [default]        OPENSSL_NO_UNIT_TEST
    uplink                 [no uplink_arch] OPENSSL_NO_UPLINK
    weak-ssl-ciphers       [default]        OPENSSL_NO_WEAK_SSL_CIPHERS
    zlib                   [default]        OPENSSL_NO_ZLIB
    zlib-dynamic           [default]        OPENSSL_NO_ZLIB_DYNAMIC
    zstd                   [default]        OPENSSL_NO_ZSTD
    zstd-dynamic           [default]        OPENSSL_NO_ZSTD_DYNAMIC
    ssl3                   [default]        OPENSSL_NO_SSL3
    ssl3-method            [default]        OPENSSL_NO_SSL3_METHOD

Config target attributes:

    AR => "ar",
    ARFLAGS => "qc",
    CC => "gcc",
    CFLAGS => "-Wall -O3",
    CXX => "g++",
    CXXFLAGS => "-Wall -O3",
    HASHBANGPERL => "/usr/bin/env perl",
    RANLIB => "ranlib",
    RC => "windres",
    asm_arch => "x86_64",
    bin_cflags => "",
    bin_lflags => "",
    bn_ops => "SIXTY_FOUR_BIT_LONG",
    build_file => "Makefile",
    build_scheme => [ "unified", "unix" ],
    cflags => "-pthread -m64",
    cppflags => "",
    cxxflags => "-std=c++11 -pthread -m64",
    defines => [ "OPENSSL_BUILDING_OPENSSL" ],
    disable => [  ],
    dso_ldflags => "-Wl,-z,defs",
    dso_scheme => "dlfcn",
    enable => [ "afalgeng" ],
    ex_libs => "-ldl -pthread",
    includes => [  ],
    lflags => "",
    lib_cflags => "",
    lib_cppflags => "-DOPENSSL_USE_NODELETE -DL_ENDIAN",
    lib_defines => [  ],
    module_cflags => "-fPIC",
    module_cxxflags => undef,
    module_ldflags => "-Wl,-znodelete -shared -Wl,-Bsymbolic",
    multilib => "64",
    perl_platform => "Unix",
    perlasm_scheme => "elf",
    shared_cflag => "-fPIC",
    shared_defflag => "-Wl,--version-script=",
    shared_defines => [  ],
    shared_ldflag => "-Wl,-znodelete -shared -Wl,-Bsymbolic",
    shared_rcflag => "",
    shared_sonameflag => "-Wl,-soname=",
    shared_target => "linux-shared",
    thread_defines => [  ],
    thread_scheme => "pthreads",
    unistd => "<unistd.h>",

Recorded environment:

    AR = 
    ARFLAGS = 
    AS = 
    ASFLAGS = 
    BUILDFILE = 
    CC = 
    CFLAGS = 
    CPP = 
    CPPDEFINES = 
    CPPFLAGS = -I/usr/local/include
    CPPINCLUDES = 
    CROSS_COMPILE = 
    CXX = 
    CXXFLAGS = 
    HASHBANGPERL = 
    LD = 
    LDFLAGS = -L/usr/local/lib
    LDLIBS = 
    MT = 
    MTFLAGS = 
    OPENSSL_LOCAL_CONFIG_DIR = 
    PERL = 
    RANLIB = 
    RC = 
    RCFLAGS = 
    RM = 
    WINDRES = 
    __CNF_CFLAGS = 
    __CNF_CPPDEFINES = 
    __CNF_CPPFLAGS = 
    __CNF_CPPINCLUDES = 
    __CNF_CXXFLAGS = 
    __CNF_LDFLAGS = 
    __CNF_LDLIBS = 

Makevars:

    AR              = ar
    ARFLAGS         = qc
    ASFLAGS         = 
    CC              = gcc
    CFLAGS          = -Wall -O3
    CPPDEFINES      = 
    CPPFLAGS        = -I/usr/local/include
    CPPINCLUDES     = 
    CXX             = g++
    CXXFLAGS        = -Wall -O3
    HASHBANGPERL    = /usr/bin/env perl
    LDFLAGS         = -L/usr/local/lib
    LDLIBS          = 
    PERL            = /usr/bin/perl
    RANLIB          = ranlib
    RC              = windres
    RCFLAGS         = 

NOTE: These variables only represent the configuration view.  The build file
template may have processed these variables further, please have a look at the
build file for more exact data:
    Makefile

build file:

    Makefile

build file templates:

    Configurations/common0.tmpl
    Configurations/unix-Makefile.tmpl
alexandre@alexandre-developpeur:~/Téléchargements/openssl$ 
`

**Comment by levitte** [comment]
> ```
> For bug reporting instructions, please see:
> <file:///usr/share/doc/gcc-13/README.Bugs>.
> ar: providers/common/der/libdefault-lib-der_rsa_sig.o: No such file or directory
> make[1]: *** [Makefile:22733: providers/libdefault.a] Error 1
> make[1]: Leaving directory '/home/alexandre/Téléchargements/openssl'
> make: *** [Makefile:3731 : build_sw] Erreur 2
> ```

I assume there were other errors before this, i.e. that `providers/common/der/libdefault-lib-der_rsa_sig.o` didn't compile for some reason.  The error that you display is a mere consequence of something going wrong when building `libdefault-lib-der_rsa_sig.o`.

Would you mind sharing the complete log of your build, so we can see all errors that occur?

**Comment by aragon5956** [comment]
i don't think that I can’t go back before : 
`alexandre@alexandre-developpeur:~/Téléchargements/openssl$ ./Configure
Configuring OpenSSL version 3.4.0-dev for target linux-x86_64
Using os-specific seed configuration
Created configdata.pm
Running configdata.pm
Created Makefile.in
Created Makefile

**********************************************************************
***                                                                ***
***   OpenSSL has been successfully configured                     ***
***                                                                ***
***   If you encounter a problem while building, please open an    ***
***   issue on GitHub <https://github.com/openssl/openssl/issues>  ***
***   and include the output from the following command:           ***
***                                                                ***
***       perl configdata.pm --dump                                ***
***                                                                ***
***   (If you are new to OpenSSL, you might want to consult the    ***
***   'Troubleshooting' section in the INSTALL.md file first)      ***
***                                                                ***
**********************************************************************
alexandre@alexandre-developpeur:~/Téléchargements/openssl$   perl configdata.pm --dump 

Command line (with current working directory = .):

    /usr/bin/perl ./Configure

Perl information:

    /usr/bin/perl
    5.36.0 for x86_64-linux-gnu-thread-multi

Enabled features:

    afalgeng
    apps
    argon2
    aria
    asm
    async
    atexit
    autoalginit
    autoerrinit
    autoload-config
    bf
    blake2
    bulk
    cached-fetch
    camellia
    capieng
    cast
    chacha
    cmac
    cmp
    cms
    comp
    ct
    default-thread-pool
    deprecated
    des
    dgram
    dh
    docs
    dsa
    dso
    dtls
    dynamic-engine
    ec
    ec2m
    ecdh
    ecdsa
    ecx
    engine
    err
    filenames
    gost
    http
    idea
    legacy
    loadereng
    makedepend
    md4
    mdc2
    module
    multiblock
    nextprotoneg
    ocb
    ocsp
    padlockeng
    pic
    pinshared
    poly1305
    posix-io
    psk
    quic
    unstable-qlog
    rc2
    rc4
    rdrand
    rfc3779
    rmd160
    scrypt
    secure-memory
    seed
    shared
    siphash
    siv
    sm2
    sm2-precomp
    sm3
    sm4
    sock
    srp
    srtp
    sse2
    ssl
    ssl-trace
    static-engine
    stdio
    tests
    thread-pool
    threads
    tls
    integrity-only-ciphers
    ts
    ui-console
    whirlpool
    tls1
    tls1-method
    tls1_1
    tls1_1-method
    tls1_2
    tls1_2-method
    tls1_3
    dtls1
    dtls1-method
    dtls1_2
    dtls1_2-method

Disabled features:

    acvp-tests             [cascade]        OPENSSL_NO_ACVP_TESTS
    asan                   [default]        OPENSSL_NO_ASAN
    brotli                 [default]        OPENSSL_NO_BROTLI
    brotli-dynamic         [default]        OPENSSL_NO_BROTLI_DYNAMIC
    buildtest-c++          [default]        
    winstore               [not-windows]    OPENSSL_NO_WINSTORE
    crypto-mdebug          [default]        OPENSSL_NO_CRYPTO_MDEBUG
    demos                  [default]        OPENSSL_NO_DEMOS
    h3demo                 [default]        OPENSSL_NO_H3DEMO
    devcryptoeng           [default]        OPENSSL_NO_DEVCRYPTOENG
    ec_nistp_64_gcc_128    [default]        OPENSSL_NO_EC_NISTP_64_GCC_128
    egd                    [default]        OPENSSL_NO_EGD
    external-tests         [default]        OPENSSL_NO_EXTERNAL_TESTS
    fips                   [default]        
    fips-securitychecks    [cascade]        OPENSSL_NO_FIPS_SECURITYCHECKS
    fuzz-afl               [default]        OPENSSL_NO_FUZZ_AFL
    fuzz-libfuzzer         [default]        OPENSSL_NO_FUZZ_LIBFUZZER
    ktls                   [default]        OPENSSL_NO_KTLS
    md2                    [default]        OPENSSL_NO_MD2 (skip crypto/md2)
    msan                   [default]        OPENSSL_NO_MSAN
    pie                    [default]        OPENSSL_NO_PIE
    rc5                    [default]        OPENSSL_NO_RC5 (skip crypto/rc5)
    sctp                   [default]        OPENSSL_NO_SCTP
    tfo                    [default]        OPENSSL_NO_TFO
    trace                  [default]        OPENSSL_NO_TRACE
    ubsan                  [default]        OPENSSL_NO_UBSAN
    unit-test              [default]        OPENSSL_NO_UNIT_TEST
    uplink                 [no uplink_arch] OPENSSL_NO_UPLINK
    weak-ssl-ciphers       [default]        OPENSSL_NO_WEAK_SSL_CIPHERS
    zlib                   [default]        OPENSSL_NO_ZLIB
    zlib-dynamic           [default]        OPENSSL_NO_ZLIB_DYNAMIC
    zstd                   [default]        OPENSSL_NO_ZSTD
    zstd-dynamic           [default]        OPENSSL_NO_ZSTD_DYNAMIC
    ssl3                   [default]        OPENSSL_NO_SSL3
    ssl3-method            [default]        OPENSSL_NO_SSL3_METHOD

Config target attributes:

    AR => "ar",
    ARFLAGS => "qc",
    CC => "gcc",
    CFLAGS => "-Wall -O3",
    CXX => "g++",
    CXXFLAGS => "-Wall -O3",
    HASHBANGPERL => "/usr/bin/env perl",
    RANLIB => "ranlib",
    RC => "windres",
    asm_arch => "x86_64",
    bin_cflags => "",
    bin_lflags => "",
    bn_ops => "SIXTY_FOUR_BIT_LONG",
    build_file => "Makefile",
    build_scheme => [ "unified", "unix" ],
    cflags => "-pthread -m64",
    cppflags => "",
    cxxflags => "-std=c++11 -pthread -m64",
    defines => [ "OPENSSL_BUILDING_OPENSSL" ],
    disable => [  ],
    dso_ldflags => "-Wl,-z,defs",
    dso_scheme => "dlfcn",
    enable => [ "afalgeng" ],
    ex_libs => "-ldl -pthread",
    includes => [  ],
    lflags => "",
    lib_cflags => "",
    lib_cppflags => "-DOPENSSL_USE_NODELETE -DL_ENDIAN",
    lib_defines => [  ],
    module_cflags => "-fPIC",
    module_cxxflags => undef,
    module_ldflags => "-Wl,-znodelete -shared -Wl,-Bsymbolic",
    multilib => "64",
    perl_platform => "Unix",
    perlasm_scheme => "elf",
    shared_cflag => "-fPIC",
    shared_defflag => "-Wl,--version-script=",
    shared_defines => [  ],
    shared_ldflag => "-Wl,-znodelete -shared -Wl,-Bsymbolic",
    shared_rcflag => "",
    shared_sonameflag => "-Wl,-soname=",
    shared_target => "linux-shared",
    thread_defines => [  ],
    thread_scheme => "pthreads",
    unistd => "<unistd.h>",

Recorded environment:

    AR = 
    ARFLAGS = 
    AS = 
    ASFLAGS = 
    BUILDFILE = 
    CC = 
    CFLAGS = 
    CPP = 
    CPPDEFINES = 
    CPPFLAGS = -I/usr/local/include
    CPPINCLUDES = 
    CROSS_COMPILE = 
    CXX = 
    CXXFLAGS = 
    HASHBANGPERL = 
    LD = 
    LDFLAGS = -L/usr/local/lib
    LDLIBS = 
    MT = 
    MTFLAGS = 
    OPENSSL_LOCAL_CONFIG_DIR = 
    PERL = 
    RANLIB = 
    RC = 
    RCFLAGS = 
    RM = 
    WINDRES = 
    __CNF_CFLAGS = 
    __CNF_CPPDEFINES = 
    __CNF_CPPFLAGS = 
    __CNF_CPPINCLUDES = 
    __CNF_CXXFLAGS = 
    __CNF_LDFLAGS = 
    __CNF_LDLIBS = 

Makevars:

    AR              = ar
    ARFLAGS         = qc
    ASFLAGS         = 
    CC              = gcc
    CFLAGS          = -Wall -O3
    CPPDEFINES      = 
    CPPFLAGS        = -I/usr/local/include
    CPPINCLUDES     = 
    CXX             = g++
    CXXFLAGS        = -Wall -O3
    HASHBANGPERL    = /usr/bin/env perl
    LDFLAGS         = -L/usr/local/lib
    LDLIBS          = 
    PERL            = /usr/bin/perl
    RANLIB          = ranlib
    RC              = windres
    RCFLAGS         = 

NOTE: These variables only represent the configuration view.  The build file
template may have processed these variables further, please have a look at the
build file for more exact data:
    Makefile

build file:

    Makefile

build file templates:

    Configurations/common0.tmpl
    Configurations/unix-Makefile.tmpl
alexandre@alexandre-developpeur:~/Téléchargements/openssl$ 
`

**Comment by levitte** [comment]
Then I suggest doing this:

``` console
$ make 2>&1 | tee makelog.txt
```

Then share `makelog.txt` with us.

**Comment by aragon5956** [comment]
for more precision the comamdn i typed was : `$ make -j12 2>&1 | tee makelog.txt`
[makelog.txt](https://github.com/user-attachments/files/16041137/makelog.txt)
Cordially

**Comment by tom-cosgrove-arm** [comment]
Left-field suggestion: could this be due to the accented characters in the directory you are running from? IIRC `Configure` puts absolute pathnames into various files, and between Perl, `make` and `gcc` there could certainly be disagreement about character sets and encodings.

Can you try in a path that only has accented characters (including the full path to `$HOME` - I don't think we see what `~` expands to in the above) and see what happens?

**Comment by levitte** [comment]
> for more precision, the command i typed was : `$ make -j12 2>&1 | tee makelog.txt` [makelog.txt](https://github.com/user-attachments/files/16041137/makelog.txt) Cordially

Did you look at it?  That shows a perfect build, no errors at all!

**Comment by nhorman** [comment]
no response from reporter, marking as inactive, to be closed at the end of 3.4 dev barring further input
